### PR TITLE
update proxy sha

### PR DESCRIPTION
**Pull request by yangminzhu**
Changes included:
```
9ca0d8f4 update istio/envoy sha (#3057)
eec9ae9f fix: implement metadata fallback on the special value (#3056)
0db11f3f use istio envoy release-1.8 instead of envoy-wasm (#3055)
58f07a77 Parse workload and service from resource labels (#3051)
33179930 grpc: add streaming metrics (#3048)
f478bcfc Automator: update common-files@master in istio/proxy@master (#3053)
cc89e662 Automator: update common-files@master in istio/proxy@master (#3052)
ddfda5e3 validate route_name (#3042)
aa356aff Remove mac test (#3050)
697c43c3 cleanup: fix TODOs waiting for c++17 (#3049)
b908a15e jwt: support nested claims (#3045)
dd64630c update envoy-wasm (upstream Envoy v1.16) (#3043)
a5ef9f7d Minor changes to Access Logs (#3000)
af4f2525 Automator: update common-files@master in istio/proxy@master (#3041)
edb74330 Add Client Side Sampling Support (#3031)
938a9485 update envoy-wasm (#3032)

```

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]
This gives us Envoy 1.16 right? Yay

**Comment by yangminzhu** [comment]
> This gives us Envoy 1.16 right? Yay

yes.

File: pilot/pkg/security/authz/model/principal.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ func principalAuthenticated(name *matcherpb.StringMatcher) *rbacpb.Principal {
 func principalSourceIP(cidr *corepb.CidrRange) *rbacpb.Principal {
 	return &rbacpb.Principal{
 		Identifier: &rbacpb.Principal_SourceIp{
+			// nolint: staticcheck
 			SourceIp: cidr,
```
Comment (howardjohn) [inline pilot/pkg/security/authz/model/principal.go:74]: @yangminzhu are you tracking updating this?

File: pilot/pkg/networking/core/v1alpha3/route/route_test.go [hunk 2]
```diff
@@ -122,6 +125,7 @@ func TestBuildHTTPRoutes(t *testing.T) {
 		g.Expect(err).NotTo(gomega.HaveOccurred())
 		g.Expect(len(routes)).To(gomega.Equal(1))
 		g.Expect(routes[0].GetRoute().Timeout.Seconds).To(gomega.Equal(int64(0)))
+		// nolint: staticcheck
 		g.Expect(routes[0].GetRoute().MaxGrpcTimeout.Seconds).To(gomega.Equal(int64(0)))
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/route/route_test.go:129]: @ramaraochavali do we just need to switch to `grpc_timeout_header_max`?

**Comment by kyessenov** [comment]
https://github.com/istio/istio/pull/27786 is broken

File: pilot/pkg/security/authz/model/principal.go (hunk 1 above)
Comment (yangminzhu) [reply pilot/pkg/security/authz/model/principal.go:74]: yes, this is updated in a follow PR in https://github.com/istio/istio/pull/27906

**Comment by yangminzhu** [comment]
blocked on https://github.com/envoyproxy/envoy/pull/13581

**Comment by bianpengyuan** [comment]
Test failure needs to be fixed. I will take over this change.

**Pull request closed without merging**